### PR TITLE
Remove default mapping

### PR DESCRIPTION
**Pull request by romseygeek**
Default mappings cannot be created in 7x indexes, so we no longer need to keep 
support for them in master.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/cluster/metadata/IndexMetaData.java [hunk 1]
```diff
@@ -1102,14 +1071,6 @@ public IndexMetaData build() {
             ImmutableOpenMap.Builder<String, AliasMetaData> tmpAliases = aliases;
             Settings tmpSettings = settings;
 
-            // update default mapping on the MappingMetaData
-            if (mappings.containsKey(MapperService.DEFAULT_MAPPING)) {
-                MappingMetaData defaultMapping = mappings.get(MapperService.DEFAULT_MAPPING);
-                for (ObjectCursor<MappingMetaData> cursor : mappings.values()) {
-                    cursor.value.updateDefaultMapping(defaultMapping);
```
Comment (jtibshirani) [inline server/src/main/java/org/elasticsearch/cluster/metadata/IndexMetaData.java]: I think we can nuke the `updateDefaultMapping` method now that this block is deleted.

**Review by jtibshirani — COMMENTED** [review]
Thanks @romseygeek, this is looking good to me! I spotted one other reference to `_default_` here: https://github.com/elastic/elasticsearch/blob/master/server/src/main/java/org/elasticsearch/action/admin/indices/mapping/get/GetMappingsResponse.java#L126

To check I'm on the same page with the upgrade plan, this is my understanding of what we will recommend to users upgrading from 7.x -> 8.x:
- There may be some indices created in 6.x that still contain default mappings. However, these will need to be reindexed before upgrading to 8.x, so we don't need any special logic to handle these mappings during the upgrade.
- If there are any index templates from 6.x that haven't been updated, users should make sure to re-add them with no default mapping (and no type).

**Comment by romseygeek** [comment]
Thanks for the review @jtibshirani (and apologies for the gap between updates).  I made some updates based on your feedback.

> There may be some indices created in 6.x that still contain default mappings. However, these will need to be reindexed before upgrading to 8.x, so we don't need any special logic to handle these mappings during the upgrade.

Right

> If there are any index templates from 6.x that haven't been updated, users should make sure to re-add them with no default mapping (and no type).

Yes - I'll check to see if we're emitting deprecation warnings here already

**Review by jtibshirani — APPROVED** [review]
This looks good to me. I see just one more reference to the default mapping that would be good to remove before merging: https://github.com/elastic/elasticsearch/blob/master/server/src/main/java/org/elasticsearch/action/admin/indices/mapping/get/GetMappingsResponse.java#L126

**Review by jpountz — APPROVED** [review]

**Comment by romseygeek** [comment]
@elasticmachine update branch

**Comment by romseygeek** [comment]
@elasticmachine update branch

**Pull request merged**